contracts/common/DelegateProxy.sol

```diff
@@ -22,7 +23,7 @@ contract DelegateProxy {
         }
     }
 
-    function isContract(address _target) constant internal returns (bool) {
+    function isContract(address _target) internal view returns (bool) {
```

Comment:
@izqui Do you know if there's a list of assembly opcodes that force a function to be `view` instead of `pure` somewhere? The [solidity docs are ambiguous](https://solidity.readthedocs.io/en/develop/contracts.html#pure-functions) as to which ones might do, but I noticed this is marked `view` (due to `extcodesize`) while the ones that deal purely with memory are marked `pure`.

Reply:
I think it is checked at compile-time @sohkai, so anything that calls `SSTORE` or `SLOAD` will error out. No runtime enforcement

Reply:
AFAIK they're not enforced at compile-time yet (see warnings in the docs).

I'm guessing any instruction that deals with on-chain storage will force `view`, so:

- `sstore`
- `sload`
- [every instruction in the "State queries" category](https://solidity.readthedocs.io/en/develop/julia.html#low-level-functions)

Reply:
Pure is for functions that dont access state (storage nor memory outside function scope)
View for functions that only read state (sload)
Functions that can modify state dont have special modifiers.